core-bundle/src/Image/Studio/FigureBuilder.php

```diff
@@ -265,6 +269,34 @@ public function from(FilesModel|ImageInterface|int|string|null $identifier): sel
         return $this->fromPath($identifier);
     }
 
+    /**
+     * Sets the image resource from a path inside a VFS storage.
+     */
+    public function fromStorage(VirtualFilesystemInterface $storage, string|Uuid $location): self
+    {
+        try {
+            $stream = $storage->readStream($location);
+        } catch (VirtualFilesystemException|UnableToResolveUuidException $e) {
+            $this->lastException = new InvalidResourceException(sprintf('Could not read resource from storage: %s', $e->getMessage()), previous: $e);
+
+            return $this;
+        }
+
+        // TODO: Our image libraries cannot deal with streams, yet. However,
```

Comment:
What exactly is the TODO here?

Reply:
After stream support is added to `contao/image`, we should remove this workaround and type restriction and directly pass on the stream to the resizer.